### ISSUE TITLE
Saved search posts aren't being counted toward the Posts per page count

### ISSUE THREAD

**Issue by realkinetix** [friendica/friendica#7840]
### Expected behavior

In my Display -> Content settings, I expect the "Number of items to display per page:" setting to do just that. If I have 20 in there, I expect 20 items to show up per page.

### Actual behavior

Possible to have an incredible number of posts show up from one's saved searches, as they don't appear to be getting counted as items being displayed in the page.

### Steps to reproduce the problem

Set the Number of items to display per page to a number, say, 20.  Save some search terms such as "#newhere", and reload the /network page.  You are likely to wind up with greater than 20 items being displayed.

### Friendica version you encountered the problem

2019.09 - 1323

### Friendica source (git, zip)

git

### PHP version

7.3

### SQL version

mariadb 10.3.17

**Comment by MrPetovan** [comment]
The number per page seems to be a general suggestion rather than a hard limit. In the opposite fashion, it's possible to have let posts per page than the limit even if you didn't reach the last page, as the posts filtered because of blocks are removing after the count is done.

**Comment by annando** [comment]
I can provide the technical background for this behaviour:
At first the system fetches the "number per page" posts. After this the system looks at the dates of the first and last post in that range of the X posts and additionally fetches all saved searches in that time range. These posts are then added to the X posts.

So when someone had subscribed to many tags but not to many persons, the system will fetch a really high number of posts.

This is something that isn't easy to change.

**Comment by realkinetix** [comment]
I guess this may be a difficult one if one is low on RAM and this kind of behaviour digs in to the file system, causing really slow and high resource searches.

Are there any other items that are implementable to help prevent some of this on the whole, such as:

- search queries are taking longer than 3 seconds - inform the user or kill the search after first result and give the user an option to bring in an older result

- set a searching time frame limit on the first page of the /network result?  If there's no results newer than 30 days or something, then inform the user with the option to search back another 30 days? (Might not be the best mechanism, but something along those lines so page 1 of /network doesn't dig in searching and searching forever...)

I don't know if any of these would be any easier to implement than the retrieval of the "right" number of results from saved searches, but I think it may be good to have some more mechanisms that prevent killer-long load times (or just massive hammering on the system's DB for what might appear to be an otherwise simple result in a user's day-to-day feed).

Hope the thoughts are useful, look forward to hearing further!

Oh, and of course, if there's anything that's already implemented that I've overlooked that I should be implementing (besides more RAM, of course, heh!) to help deal with this, please hit me with the clue bat here or back on the F* forum.  Maybe I have an obscure setting off somewhere that's making my situation worse.

**Comment by annando** [comment]
We would had to perform these SQL queries asynchronously to be able to stop them. This would be really complicated.

**Comment by realkinetix** [comment]
OK, was just throwing out some suggestions... considering the load it can place on systems, is this something that is worth working on?

**Comment by MrPetovan** [comment]
I personally think saved searches should be saved searches and not hashtag follow which it definitely doesn't advertise it is.

**Comment by annando** [comment]
So we should rename it.

**Comment by AlfredSK** [comment]
Rename to what? It's already called 'Saved Searches' in the left aside.

**Comment by MrPetovan** [comment]
Rename to "Followed hashtags"

**Comment by annando** [comment]
Exactly. Some time ago we extended the previous "saved searches" with the functionality to the current behavior of following the hashtags.

**Comment by AlfredSK** [comment]
But it does more than just 'follow hashtags'. You can save every search not only hashtag searches. Or am I wrong?

**Comment by annando** [comment]
The original issue had been solved a long time ago.
